tests/console/verify_secure_boot.pm

```diff
@@ -17,13 +17,17 @@ use Test::Assert ':all';
 sub run {
     my $test_data = get_test_suite_data();
     select_console 'root-console';
-    record_info("Check file", "Check if file /sys/firmware/efi/vars/SecureBoot-*/data exists");
-    assert_script_run("ls /sys/firmware/efi/vars/SecureBoot-*/data");
+    record_info("Check file", "Check if file /sys/firmware/efi/efivars/SecureBoot-* exists");
+    # From v6.0 kernel and onwards, the SecureBoot file resides in /sys/firmware/efi/efivars/
+    assert_script_run("ls /sys/firmware/efi/efivars/SecureBoot-*");
     record_info("Check secure boot", "Check if secure boot option is set as expected");
-    my $secure_boot = script_output("od -An -t u1 /sys/firmware/efi/vars/SecureBoot-*/data") ? 'enabled' : 'disabled';
+    my $octal_str = script_output("od -An -t u1 /sys/firmware/efi/efivars/SecureBoot-*");
+
+    # The final octal digit shows whether Secure Boot is enabled (1) or disabled (0)
+    my $secure_boot_octal = substr($octal_str, -1);
+    die 'Unexpected values in Secure Boot file' unless (length($octal_str) == 17) and ($secure_boot_octal =~ /0|1/);
```

Comment:
About line 27: still we are doing some kind of indexing, we have a library to get the first item in Perl, I don't remember the name right now, probably has the option to get the last too..
About line 28: do we know that 17 is a fixed number, is somewhere documented? perhaps those spaces between the real values can change from 3 to 2 in next version and the validation breaks. We could still split them by regex as mentioned, considering that could be 1 or 3 or 5 spaces let's say if we don't know it is quite safe like that.